### DISCUSSION TITLE
Zigbee2Mqtt plugin

### DISCUSSION THREAD

**Discussion by adorobis** [tprelog/iocage-homeassistant#49] [Ideas]
@tprelog You are doing great work with the plugins for TrueNAS, Many thanks for that and great support you are providing! How about creating one for zigbee2mqtt?

**Comment by tprelog** [comment]
> How about creating one for zigbee2mqtt?

Already done by @paxswill - See the following repository for details

https://github.com/paxswill/iocage-zigbee2mqtt

**Reply by adorobis** [reply]
Ah, great, thanks! So far I've installed manually, facing a bit of problem with latest node.js which I could not make work in 12.2 RELEASE jail so I've installed node14 instead.
I have a different question though: is there a way to reference to a USB device in a more static way? Sometimes after system restart or after disconnecting and reconnecting USB device again the number in the /dev/cuaU[number] changes so I have to change the config of zigbee and zwave modules.

**Reply by tprelog** [reply]
Sorry, I just saw this...

> I have a different question though: is there a way to reference to a USB device in a more static way?

on SCALE, yes - But I'm not aware of anything for FreeBSD or TrueNAS CORE

**Reply by adorobis** [reply]
I have found a way to make it sort of static. It is a script that it creating a symbolic link to a static name upon a jail start. Below example for my zwave and zigbee sticks. I had to split that into separate scripts for each of the jails as sometimes it was conflicting with each other, below in a single script for both sticks and jails:
```
#!/bin/bash

function get_tty () {
    local vendor="${1}"
    local product="${2}"
    # return the tty{} value, eg; U2
    sysctl dev.umodem | grep "vendor=${vendor} product=${product}" | sed -r 's/.*ttyname=([^\s]+) .*/\1/'
}

function create_symlink () {
    local source="${1}"
    # failsafe
    if [ "${source}" == 'tty' ]; then return; fi
    local target="/mnt/[pool_name]/iocage/jails/${3}/root/dev/${2}"
    if [ -e "${target}" ]; then rm -f "${target}"; fi
    ln -s "${source}" "${target}"
}

# zigbee is a Conbee 2
create_symlink "cua$(get_tty '0x1cf1' '0x0030')" "cuaUzigbee" "[zigbee_jail]"
# zwave is a Z-Stick Gen 5
create_symlink "cua$(get_tty '0x0658' '0x0200')" "cuaUzwave" "[zwave_jail]"
```
